src/ngraph/runtime/interpreter/int_backend.cpp

```diff
@@ -159,42 +153,35 @@ bool runtime::interpreter::INTBackend::call(shared_ptr<Function> function,
         {
             continue;
         }
-        if (type_id == OP_TYPEID::Constant)
-        {
-            const op::Constant* c = static_cast<const op::Constant*>(op);
-            descriptor::Tensor* tensor = op->get_output_tensor_ptr(0).get();
-            tensor_map.insert({tensor, const_cast<void*>(c->get_data_ptr())});
-            continue;
-        }
+
         // get op inputs from map
-        vector<const void*> op_inputs;
+        vector<shared_ptr<HostTensor>> op_inputs;
         for (const descriptor::Input& input : op->get_inputs())
         {
             descriptor::Tensor* tensor = input.get_output().get_tensor_ptr().get();
             op_inputs.push_back(tensor_map.at(tensor));
         }
 
         // get op outputs from map or create
-        vector<void*> op_outputs;
-        vector<shared_ptr<runtime::HostTensor>> htv_outputs;
+        vector<shared_ptr<HostTensor>> op_outputs;
         for (size_t i = 0; i < op->get_output_size(); ++i)
         {
             descriptor::Tensor* tensor = op->get_output_tensor_ptr(i).get();
-            void* host_tensor = nullptr;
+            shared_ptr<HostTensor> host_tensor;
             auto it = tensor_map.find(tensor);
             if (it == tensor_map.end())
             {
-                auto offset = op->get_output_tensor(i).get_pool_offset();
-                host_tensor = instance.get_temporary_pointer(offset);
+                const Shape& shape = op->get_output_shape(i);
```

Comment:
For dynamic ops, the shape (and the element type, actually) can be dynamic at this point, and this is going to fail (assert) in that case. I think we'll either need to rig up a mechanism to delay creation of the `HostTensor` until runtime when the full shape is known, or make it possible for the `HostTensor` itself to have dynamic shape and delay allocation and/or allow resizing of its underlying buffer.

Reply:
Can we address this issue in a separate PR as it is beyond the scope of this PR. This PR is only to change the INTERPRETER from using a memory pool approach to using on-demand temporary tensor allocation.

Reply:
Sounds good!